### DISCUSSION TITLE
How to change the permitted length of the @Fields.string()

### DISCUSSION THREAD

**Discussion by SubliminalGuy** [noam-honig/radweb#233] [Q&A]
Hi everyone,

in one of my apps i'm initiating a String Field with 
```js
@Fields.string()
  sprechertext = "";
```
When i'm trying to POST a string above a certain threshold i'm receiving an error from the backend:

"Data too long for column 'sprechertext' at row 1"

Is there an implicit length constraint on the string field? What can I do, to process longer strings?

Thanks for your help!

**Comment by noam-honig** [comment]
Hi @SubliminalGuy , The limitation is coming from the underlying database, not from remult.

You can determine the text field length that will be sent to the database on the create script, using the `maxLength` option:
```ts
@Fields.string({maxLength:200})
```

If you want to use a different datatype when you create the table, you can specify that data type in the valueConverter property:
```ts
Fields.string({valueConverter:{
    fieldTypeInDb:'varchar(max)'
}})
```

Note that `varchar(max)` is relevant to the syntax of the specific database.

Let me know if this helps

**Reply by SubliminalGuy** [reply]
Yes, that was exactly what is was looking for! Thanks a lot! :-)
